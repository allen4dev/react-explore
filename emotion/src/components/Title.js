import React from 'react';
import { string } from 'prop-types';
import styled from '@emotion/styled';

const Wrapper = styled.h1`
  color: white;
`;

function Title({ children, ...props }) {
  return (
    <Wrapper {...props} data-testid="title">
      {children}
    </Wrapper>
  );
}

Title.propTypes = {
  children: string.isRequired,
};

export default Title;
