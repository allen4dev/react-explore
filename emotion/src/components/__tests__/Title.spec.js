import React from 'react';
import { render } from 'react-testing-library';

import Title from '../Title';

it('should render', () => {
  const { container } = render(<Title>My title</Title>);

  expect(container.firstChild).toMatchSnapshot();
});
