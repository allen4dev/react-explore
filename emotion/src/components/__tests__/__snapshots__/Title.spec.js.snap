// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
.emotion-0 {
  color: white;
}

<h1
  class="emotion-0 emotion-1"
  data-testid="title"
>
  My title
</h1>
`;
